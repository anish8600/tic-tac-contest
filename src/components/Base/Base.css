.base-container{
    margin-left: 5px;
}