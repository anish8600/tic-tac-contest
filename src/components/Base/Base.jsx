import React from 'react';
import PropTypes from 'prop-types';
import TicTac from '../TicTac/TicTac';

class Base extends React.Component{
  formMatrix(i){
    return (
      <TicTac value={this.props.baseArray[i]} onClick={() => this.props.onClick(i)}/>
    );
  }
  
  render() {
      return (
        <div>
          <div className="base">
            {this.formMatrix(0)}
            {this.formMatrix(1)}
            {this.formMatrix(2)}
          </div>
          <div className="base">
            {this.formMatrix(3)}
            {this.formMatrix(4)}
            {this.formMatrix(5)}
          </div>
          <div className="base">
            {this.formMatrix(6)}
            {this.formMatrix(7)}
            {this.formMatrix(8)}
          </div>
        </div>
      );
    }
}

Base.propTypes = { baseArray: PropTypes.array.isRequired, onClick: PropTypes.func.isRequired};

export default Base;