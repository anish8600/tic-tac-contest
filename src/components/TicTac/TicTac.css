.tictac {
    height: 4em;
    width: 4em;
    margin-right: -1px;
    margin-top: -1px;
    font-size: 2em;
    font-weight: bold;
    line-height: 2em;
    padding: 0px;
    text-align: center;
    float: left;
    border: 1px solid #999;    
  }