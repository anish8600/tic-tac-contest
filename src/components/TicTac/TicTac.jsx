import React from 'react';
import PropTypes from 'prop-types';
import './TicTac.css';

const TicTac = (props) => <button className="tictac" onClick={props.onClick}> {props.value} </button>

TicTac.propTypes = { value: PropTypes.string, onClick: PropTypes.func.isRequired};

export default TicTac;