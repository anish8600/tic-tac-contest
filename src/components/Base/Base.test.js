import React from 'react';
import { shallow } from 'enzyme';
import Base from './Base';
import TicTac from '../TicTac/TicTac';

describe('Base', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Base baseArray={[]} onClick={jest.fn()} />));

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(4);
  });

  xit('should render the TicTac Component', () => {
    expect(wrapper.containsMatchingElement(<TicTac value={''} onClick={()=>{}}/>)).toEqual(true);
  });

});