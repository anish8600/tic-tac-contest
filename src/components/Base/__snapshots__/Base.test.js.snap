// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Base
    baseArray={Array []}
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="base"
>
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
</div>,
        <div
          className="base"
>
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
</div>,
        <div
          className="base"
>
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
          <TicTac
                    onClick={[Function]}
                    value={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
          ],
          "className": "base",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
          ],
          "className": "base",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
            <TicTac
              onClick={[Function]}
              value={undefined}
/>,
          ],
          "className": "base",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="base"
>
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
</div>,
          <div
            className="base"
>
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
</div>,
          <div
            className="base"
>
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
            <TicTac
                        onClick={[Function]}
                        value={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
            ],
            "className": "base",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
            ],
            "className": "base",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
              <TicTac
                onClick={[Function]}
                value={undefined}
/>,
            ],
            "className": "base",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
